### PR TITLE
sql: periodically refresh table leases

### PR DESCRIPTION
**Pull request by vivekmenezes**
This change periodically refreshes some of the table leases.
The current limit is 50 tables and can be configured using
sql.tablecache.lease.refresh_limit

This change will eventually be replaced by epoch based
table leases

related to #23510

Release note (sql change): Fix problem with needing to run
a periodic sql query on the outside to get good initial
latency on a dormant cluster.

### CODE REVIEW

**Comment by vivekmenezes** [comment]
I'll add in a unittest before we merge this.

File: pkg/sql/lease.go
Comment (dt) [inline pkg/sql/lease.go]: I think we want to see this with some initial IDs on boot to avoid the "cold start" problem -- though as-is, this should help with the "cooling off".

My prior PR iterated the descriptors table to get IDs: https://github.com/cockroachdb/cockroach/pull/27017/files

Comment (vivekmenezes) [reply pkg/sql/lease.go]: I'm not too fond of that approach. Let's keep this and introduce a fix for cold start separately. Hold off on reviewing further until I have a test. Thanks!

**Review by dt — APPROVED** [review]

**Pull request merged**
